server/coordinator.go

```diff
@@ -107,6 +109,16 @@ func (c *coordinator) dispatch(region *core.RegionInfo) {
 			c.removeOperator(op)
 		}
 	}
+
+	// Check merge operator.
+	if c.limiter.OperatorCount(schedule.OpMerge) >= c.cluster.GetMergeScheduleLimit() {
+		return
+	}
+
+	if op1, op2 := c.mergeChecker.Check(region); op1 != nil && op2 != nil {
```

Comment:
move it into `patroRegions()`. We don't create operators when receiving region heartbeat any more.